clang/lib/DPCT/IncMigration/ReMigration.cpp

```diff
@@ -0,0 +1,777 @@
+convertReplcementsLineString(const std::vector<Replacement> &InRepls) {
+  std::vector<Replacement> Replacements =
+      splitReplInOrderToNotCrossLines(InRepls);
+  UnifiedPath FilePath(InRepls[0].getFilePath());
+
+  // group replacement by line
+  std::map<unsigned, std::vector<Replacement>> ReplacementsByLine;
+  for (const auto &Repl : Replacements) {
+    unsigned LineNum = getLineNumber(FilePath, Repl.getOffset());
+    ReplacementsByLine[LineNum].push_back(Repl);
+  }
+
+  // process each line
+  std::map<unsigned, std::string> Result;
+  for (auto &[LineNum, Repls] : ReplacementsByLine) {
+    std::sort(Repls.begin(), Repls.end());
+
+    std::string OriginalLineStr = getLineString(FilePath, LineNum).str();
+    unsigned LineStartOffset = getLineBeginOffset(FilePath, LineNum);
+    std::string NewLineStr;
+    unsigned Pos = 0;
+    for (const auto &Repl : Repls) {
+      unsigned StrOffset = Repl.getOffset() - LineStartOffset;
+      NewLineStr += OriginalLineStr.substr(Pos, StrOffset - Pos);
+      NewLineStr += Repl.getReplacementText().str();
+      Pos = StrOffset + Repl.getLength();
+    }
+    NewLineStr += OriginalLineStr.substr(Pos);
+    Result[LineNum] = NewLineStr;
+  }
+  return Result;
+}
+
+static std::map<unsigned, std::string>
+convertReplcementsLineString(const tooling::Replacements &Repls) {
+  std::vector<Replacement> ReplsVec;
+  for (const auto &R : Repls) {
+    ReplsVec.push_back(R);
+  }
+  return convertReplcementsLineString(ReplsVec);
+}
+
+std::vector<Replacement>
+mergeMapsByLine(const std::map<unsigned, std::string> &MapA,
+                const std::map<unsigned, std::string> &MapB,
+                const UnifiedPath &FilePath) {
+  auto genReplacement = [&](unsigned LineNumber,
+                            const std::string &LineContent) {
+    unsigned Offset = getLineBeginOffset(FilePath, LineNumber);
+    unsigned StrLen = getLineString(FilePath, LineNumber).size();
+    return Replacement(FilePath.getCanonicalPath(), Offset, StrLen,
+                       LineContent);
+  };
+
+  std::vector<Replacement> Result;
+  auto ItA = MapA.begin();
+  auto ItB = MapB.begin();
+
+  while (ItA != MapA.end() || ItB != MapB.end()) {
+    if (ItA == MapA.end()) {
+      Result.push_back(genReplacement(ItB->first, ItB->second));
+      ++ItB;
+    } else if (ItB == MapB.end()) {
+      Result.push_back(genReplacement(ItA->first, ItA->second));
+      ++ItA;
+    } else if (ItA->first < ItB->first) {
+      Result.push_back(genReplacement(ItA->first, ItA->second));
+      ++ItA;
+    } else if (ItB->first < ItA->first) {
+      Result.push_back(genReplacement(ItB->first, ItB->second));
+      ++ItB;
+    } else {
+      // ItB->first == ItA->first
+      if (ItB->second == ItA->second) {
+        // No conflict, just add one of them
+        Result.push_back(genReplacement(ItA->first, ItA->second));
+        ++ItA;
+        ++ItB;
+        continue;
+      }
+      // Conflict line(s)
```

Comment:
We can do some refinements here.
If Repl_B and Repl_D have a conflict at one line, and no user-change in this line, we can user Repl_B directly.